create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusConverter.java

```diff
@@ -163,4 +177,62 @@ private AdditionalIdentifier toAdditionalIdentifier(ItemidTp itemIdTp) {
         return new AdditionalIdentifier(ScopusConstants.ADDITIONAL_IDENTIFIERS_SCOPUS_ID_SOURCE_NAME,
                                         itemIdTp.getValue());
     }
+
+    private PublicationContext getPublicationContext() {
+        if (isJournal(docTp)) {
+            return attempt(() -> createUnconfirmedJournal(docTp))
+                    .orElseThrow(fail -> logErrorAndThrowException(fail.getException()));
+        }
+        return ScopusConstants.EMPTY_PUBLICATION_CONTEXT;
+    }
+
+    private RuntimeException logErrorAndThrowException(Exception exception) {
+        logger.error(exception.getMessage());
+        return exception instanceof RuntimeException
+                ? (RuntimeException) exception
+                : new RuntimeException(exception);
+    }
+
+    private UnconfirmedJournal createUnconfirmedJournal(DocTp docTp) throws InvalidIssnException {
+        var source = getSource(docTp);
```

Comment:
remove all parameters where the parameter is the field. e.g. getSource(),isJournal(), etc.